tests/build_parents_file.sh

```diff
@@ -0,0 +1,251 @@
+fi
+
+get_parent_version() {
+    devfile=$1
+    name=$2
+    version=$($YQ_PATH eval .parent.version ${devfile})
+
+    if [ "${version}" == "null" ] && [ -f "${stacks_dir}/${name}/stack.yaml" ]; then
+        version=$($YQ_PATH eval '.versions | filter(.default) | .[0].version' ${stacks_dir}/${name}/stack.yaml)
+    fi
+
+    echo ${version}
+}
+
+# Get parent index if exists, else returns -1
+parent_index() {
+    name=$1
+    version=$2
+
+    if [ -z "${version}" ]; then
+        result=$($YQ_PATH eval ".parents | to_entries | filter(.value.name == \"${name}\") | .[0].key" ${parents_file})
+    else
+        result=$($YQ_PATH eval ".parents | to_entries | filter(.value.name == \"${name}\" and .value.version == \"${version}\") | .[0].key" ${parents_file})
+    fi
+
+    if [ "${result}" == "null" ]; then
+        echo "-1"
+    else
+        echo ${result}
+    fi
+}
+
+# Get child index if exists, else returns -1
+child_index() {
+    parent_idx=$1
+    name=$2
+    version=$3
+
+    if [ -z "${version}" ]; then
+        result=$($YQ_PATH eval ".parents.[${parent_idx}].children | to_entries | filter(.value.name == \"${name}\") | .[0].key" ${parents_file})
+    else
+        result=$($YQ_PATH eval ".parents.[${parent_idx}].children | to_entries | filter(.value.name == \"${name}\" and .value.version == \"${version}\") | .[0].key" ${parents_file})
+    fi
+
+    if [ "${result}" == "null" ]; then
+        echo "-1"
+    else
+        echo ${result}
+    fi
+}
+
+# Builds sample parents
+build_parents() {
+    parent_name=$1
+    parent_version=$2
+
+    if [ "${parent_version}" == "null" ]; then
+        parent_version=""
+    fi
+
+    if [ "${parent_name}" != "null" ]; then
+        if [ ! -f ${parents_file} ]; then
+            $YQ_PATH eval -n ".parents[0].name = \"${parent_name}\"" > ${parents_file}
+            if [ "${parent_version}" != "" ]; then
+                $YQ_PATH eval ".parents[0].version = \"${parent_version}\"" -i ${parents_file}
+            fi
+            
+            return
+        fi
+
+        if [ "$($YQ_PATH eval .parents ${parents_file})" == "null" ]; then
+            $YQ_PATH eval ".parents[0].name = \"${parent_name}\"" -i ${parents_file}
+            if [ "${parent_version}" != "" ]; then
+                $YQ_PATH eval ".parents[0].version = \"${parent_version}\"" -i ${parents_file}
+            fi
+
+            return
+        fi
+
+        parent_idx=$(parent_index ${parent_name} ${parent_version})
+        if [ "${parent_idx}" == "-1" ]; then
+            next_idx=$($YQ_PATH eval ".parents | length" ${parents_file})
+            $YQ_PATH eval ".parents[${next_idx}].name = \"${parent_name}\"" -i ${parents_file}
+            if [ "${parent_version}" != "" ]; then
+                $YQ_PATH eval ".parents[${next_idx}].version = \"${parent_version}\"" -i ${parents_file}
+            fi
+        fi
+    else
+        return 1
+    fi
+}
+
+# Builds children of parent stacks
+build_children() {
```

Comment:
I was thinking if it makes sense to rename this to `build_child` (singular) as (if I'm not wrong) we only build a specific sample version.